ipywidgets/static/widgets/js/widget_box.js

```diff
@@ -16,8 +16,74 @@ define([
         }, widget.WidgetModel.serializers)
     });
 
+    var ProxyModel = widget.WidgetModel.extend({}, {
+        serializers: _.extend({
+            child: {deserialize: widget.unpack_models},
+        }, widget.WidgetModel.serializers),
+    });
+
+    var ProxyView = widget.WidgetView.extend({
+        initialize: function() {
+            // Public constructor
+            ProxyView.__super__.initialize.apply(this, arguments);
+            this.$box = this.$el;
+            this.child_promise = Promise.resolve();
+        },
+
+        render: function() {
+            var that = this;
+            var child_view = this.set_child(this.model.get("child"))
+            this.listenTo(this.model, "change:child", function(model, value) {
+                this.set_child(value);
+            });
+            return child_view;
+        },
+
+        remove: function() {
+            ProxyView.__super__.remove.apply(this, arguments);
+            this.child_promise.then(this.child.remove);
```

Comment:
You also need to check that child is not undefined, because child_promise, misleadingly, does not guarantee a child has been set.